### PR TITLE
Add null-safety checks in Splat and SplatOverlay

### PR DESCRIPTION
**Pull request by slimbuck**
## Summary
- Add optional chaining (`?.`) to `gsplat` access in `SplatOverlay.detach()` to prevent crashes when the gsplat component has already been removed before detach is called
- Add optional chaining (`?.`) to `scene` access in `Splat.visible` setter to prevent crashes when visibility is set during element destruction (after scene reference is cleared)

## Details
Both fixes guard against null reference errors that can occur during teardown/cleanup sequences where the order of destruction isn't guaranteed. Without these guards, accessing `.gsplat.instance` or `.events.fire()` on a null reference throws a runtime error.

### CODE REVIEW

**Review by willeastcott — APPROVED** [review]

File: src/splat-overlay.ts [hunk 1]
```diff
@@ -142,7 +142,7 @@ class SplatOverlay extends Element {
     detach() {
         // unsubscribe from sorter updates
         if (this.splat && this.onSorterUpdated) {
-            this.splat.entity.gsplat.instance.sorter.off('updated', this.onSorterUpdated);
+            this.splat.entity.gsplat?.instance.sorter.off('updated', this.onSorterUpdated);
```
[the remark this answers is not in the document]
Comment (slimbuck) [reply src/splat-overlay.ts]: Fixed -- extended optional chaining through `instance` and `sorter` as well: `gsplat?.instance?.sorter?.off(...)`
